### PR TITLE
flavors: Make ShortNames for flavors shorter

### PR DESCRIPTION
**Pull request by lyarwood**
**What this PR does / why we need it**:

This PR shortens the flavor and clusterflavor CRD shortnames so that they follow the existing examples VirtualMachine with vm and VirtualMachineInstance with vmi.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

N/A

**Special notes for your reviewer**:

N/A

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/resource/generate/components/crds.go [hunk 1]
```diff
@@ -531,7 +531,7 @@ func NewVirtualMachineFlavorCrd() (*extv1.CustomResourceDefinition, error) {
 		Names: extv1.CustomResourceDefinitionNames{
 			Plural:     flavor.PluralResourceName,
 			Singular:   flavor.SingularResourceName,
-			ShortNames: []string{"vmflavor", "vmflavors"},
```
Comment (xpivarc) [inline pkg/virt-operator/resource/generate/components/crds.go:534]: IMHO this is changing the API. I would just add the new shorter names. WDYT?

**Review by xpivarc — CHANGES_REQUESTED** [review]
/hold
I think we don't want to break any automation.

File: pkg/virt-operator/resource/generate/components/crds.go (hunk 1 above)
Comment (lyarwood) [reply pkg/virt-operator/resource/generate/components/crds.go:534]: I'd agree if the API itself wasn't v1alpha1 at the moment, I also couldn't find any examples of automation within the kubevirt namespace using these. At this point I think we are still okay renaming shortnames.

FWIW this is just some earlier refactoring ahead of a much larger series based on my design proposal for flavors and preferences [1] that's going to introduce many more breaking changes to this API.

[1] https://github.com/kubevirt/community/pull/165

Comment (lyarwood) [reply pkg/virt-operator/resource/generate/components/crds.go:534]: Actually that's fine but I'd still like to remove them before this API goes hits beta.

**Comment by lyarwood** [comment]
@xpivarc @akrejcir would you mind hitting this again please?

File: pkg/virt-operator/resource/generate/components/crds.go (hunk 1 above)
Comment (xpivarc) [reply pkg/virt-operator/resource/generate/components/crds.go:534]: Right. By automation, I thought any automation outside of org. This reminds me that we should merge  https://github.com/kubevirt/community/pull/160

**Pull request merged**